libpromises/eval_context.c

```diff
@@ -1443,6 +1450,8 @@ static bool EvalContextClassPut(EvalContext *ctx, const char *ns, const char *na
         return false;
     }
 
+    StringSetAdd(ctx->all_classes, SafeStringDuplicate(name));
```

Comment:
Hmmm I see what you mean; This is not trivial, but I'll try to think of something since the memory overhead is significant I think.

BTW, do you store the fully qualified name in the set? i.e. the namespace and the scope? Or just the class name?